src/main/javascript/monet.d.ts

```diff
@@ -303,20 +303,30 @@ declare namespace monet {
      */
 
     interface IO<T> extends IMonad<T> {
+        /* Inherited from Monad: */
         bind<V>(fn: (val: T) => IO<V>): IO<V>;
         flatMap<V>(fn: (val: T) => IO<V>): IO<V>;
         chain<V>(fn: (val: T) => IO<V>): IO<V>;
-        map<V>(fn: (val: T) => V): IO<V>;
-        run(): void;
-        perform(): void;
+        map<V>(fn: (v: T) => V): IO<V>;
         join<V>(): IO<V>; // if T is IO<V>
         takeLeft(m: IO<T>): IO<T>;
         takeRight(m: IO<T>): IO<T>;
+
+        /* Inherited from Applicative: */
+        ap<V>(ioFn: IO<(v: T) => V>): IO<V>;
+
+        /* IO specific: */
+        run(): T;
+        perform(): T;         // Alias for run()
+        performUnsafeIO(): T; // Alias for run()
     }
 
     interface IIOStatic {
         <T>(fn: () => T): IO<T>;
-        new <T>(fn: () => T): IO<T>;
+        unit: IIOStatic;
```

Comment:
I'm not sure about the decision to type them all as `I<X>Static`. This will make IDE's happily suggest things like `IO.unit.pure.io.of.pure.unit(() => console.log("x"))`.